session.go

```diff
@@ -418,11 +422,28 @@ func (s *session) WriteBytes(pkg []byte) (int, error) {
 		return 0, ErrSessionClosed
 	}
 
-	lg, err := s.Connection.send(pkg)
+	leftPackageSize, totalSize, writeSize := len(pkg), len(pkg), 0
+	s.wlock.Lock()
```

Comment:
In my opinion, the wlock used for protect big package send, so may be u can change the wlock to rwMutex, and only when package size bigger than 16kb, try to get Lock, other times and other places you can use Rlock.

Reply:
good idea

Reply:
address